apps/jobs/views.py

```diff
@@ -1,3 +1,87 @@
+from django.conf import settings
 from django.shortcuts import render
 
-# Create your views here.
+from rest_framework import permissions, status
+from rest_framework.decorators import (api_view,
+                                       authentication_classes,
+                                       permission_classes,
+                                       throttle_classes,)
+
+from rest_framework_expiring_authtoken.authentication import (
+    ExpiringTokenAuthentication,)
+from rest_framework.response import Response
+from rest_framework.throttling import UserRateThrottle
+
+from accounts.permissions import HasVerifiedEmail
+from challenges.models import (
+    ChallengePhase,
+    Challenge,)
+from participants.models import (ParticipantTeam, Participant)
+from participants.utils import (
+    get_participant_team_id_of_a_user_for_a_challenge,
+    check_user_participated_in_challenge,)
+
+from .models import Submission
+
+
+@throttle_classes([UserRateThrottle])
+@api_view(['POST'])
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail))
+@authentication_classes((ExpiringTokenAuthentication,))
+def challenge_submission(request, challenge_id, challenge_phase_id):
+    """API Endpoint for making a submission to a challenge"""
+    # check if the challenge exists or not
+    try:
+        challenge = Challenge.objects.get(pk=challenge_id)
+    except Challenge.DoesNotExist:
+        response_data = {'error': 'Challenge does not exist!'}
```

Comment:
Lets remove the exclamation mark. We have been following that convention so far.